tools/cold-store/src/cli.rs

```diff
@@ -57,3 +83,93 @@ fn print_heads(store: &NodeStorage) {
         store.get_store(Temperature::Cold).get_ser::<Tip>(DBCol::BlockMisc, HEAD_KEY)
     );
 }
+
+fn copy_next_block(store: &NodeStorage, config: &NearConfig, hot_runtime: &Arc<NightshadeRuntime>) {
+    // Cold HEAD can be not set in testing.
+    // It should be set before the copying of a block in prod,
+    // but we should default it to genesis height here.
+    let cold_head_height = store
+        .get_store(Temperature::Cold)
+        .get_ser::<Tip>(DBCol::BlockMisc, HEAD_KEY)
+        .unwrap_or_else(|e| panic!("Error reading cold HEAD: {:#}", e))
+        .map_or(config.genesis.config.genesis_height, |t| t.height);
+
+    // If FINAL_HEAD is not set for hot storage though, we default it to 0.
+    // And subsequently fail in assert!(next_height <= hot_final_height).
+    let hot_final_head = store
+        .get_store(Temperature::Hot)
+        .get_ser::<Tip>(DBCol::BlockMisc, FINAL_HEAD_KEY)
+        .unwrap_or_else(|e| panic!("Error reading hot FINAL_HEAD: {:#}", e))
+        .map(|t| t.height)
+        .unwrap_or(0);
+
+    let next_height = cold_head_height + 1;
+    println!("Height: {}", next_height);
+    assert!(next_height <= hot_final_head, "Should not copy non final blocks");
+
+    // Here it should be sufficient to just read from hot storage.
+    // Because BlockHeight is never garbage collectable and is not even copied to cold.
+    let cold_head_hash = get_ser_from_store::<CryptoHash>(
+        store,
+        Temperature::Hot,
+        DBCol::BlockHeight,
+        &cold_head_height.to_le_bytes(),
+    ).unwrap_or_else(||panic!("No block hash in hot storage for height {}", cold_head_height));
+
+    // For copying block we need to have shard_layout.
+    // For that we need epoch_id.
+    // For that we might use prev_block_hash, and because next_hight = cold_head_height + 1,
+    // we use cold_head_hash.
+    update_cold_db(
+        &*store.cold_db().unwrap(),
```

Comment:
Just out of curiosity what does the &* do? It seems like those two should cancel each other.